### ISSUE TITLE
js/require not available after loading

### ISSUE THREAD

**Issue by chriswatrous** [thheller/shadow-cljs#528]
If I start a repl with `npx shadow-cljs cljs-repl app`, js/require works.

If I put js/require in a cljs file so that it runs when the namespace loads, it works.

If I try to evaluate js/require from my editor (Atom with proto-repl) connected to the app, it gives this error:
```
:1
require("fs")
^

ReferenceError: require is not defined
    at :1:1
    at Script.runInThisContext (vm.js:119:20)
    at runInThisContext (vm.js:326:38)
    at global.SHADOW_NODE_EVAL (/Users/chris/projects/clojure/shadow-bug/target/main.js:81:30)
    at Object.shadow$cljs$devtools$client$node$node_eval [as node_eval] (/Users/chris/projects/clojure/shadow-bug/.shadow-cljs/builds/app/dev/out/cljs-runtime/shadow/cljs/devtools/client/node.cljs:24:1)
    at /Users/chris/projects/clojure/shadow-bug/.shadow-cljs/builds/app/dev/out/cljs-runtime/shadow/cljs/devtools/client/node.cljs:48:13
    at Object.shadow$cljs$devtools$client$env$repl_call [as repl_call] (/Users/chris/projects/clojure/shadow-bug/.shadow-cljs/builds/app/dev/out/cljs-runtime/shadow/cljs/devtools/client/env.cljs:102:11)
    at Object.shadow$cljs$devtools$client$node$repl_invoke [as repl_invoke] (/Users/chris/projects/clojure/shadow-bug/.shadow-cljs/builds/app/dev/out/cljs-runtime/shadow/cljs/devtools/client/node.cljs:46:1)
    at shadow$cljs$devtools$client$node$process_message (/Users/chris/projects/clojure/shadow-bug/.shadow-cljs/builds/app/dev/out/cljs-runtime/shadow/cljs/devtools/client/node.cljs:116:5)
    at Object.shadow$cljs$devtools$client$env$process_ws_msg [as process_ws_msg] (/Users/chris/projects/clojure/shadow-bug/.shadow-cljs/builds/app/dev/out/cljs-runtime/shadow/cljs/devtools/client/env.cljs:164:7)
```

If I eval `(js/Object.keys js/global)`, require is not there.

`(js/console.log js/global.require)` prints `undefined`.

If I save the function like this: `(def js-require js/require)`, I can call it with Atom/proto-repl.

**Comment by thheller** [comment]
Which `:target` is the `:app` build using?

**Comment by chriswatrous** [comment]
:node-script

**Comment by chriswatrous** [comment]
Edited original comment to change repl command to `npx shadow-cljs node-repl app`. I'm pretty sure that's what I originally used.

**Comment by thheller** [comment]
I can use either form just fine in a node-repl?

```
$ shadow-cljs node-repl
shadow-cljs - config: /mnt/c/Users/thheller/code/shadow-cljs/shadow-cljs.edn  cli version: 2.8.40  node: v10.13.0
shadow-cljs - connected to server
[1:1]~cljs.user=> (js/require "fs")
#js {:appendFile #object[appendFile], :appendFileSync #object[appendFileSync], :access #object[access], :accessSync #object[accessSync], :chown #object[chown], :chownSync #object[chownSync], :chmod #object[chmod], :chmodSync #object[chmodSync], :close #object[close], :closeSync #object[closeSync], :copyFile #object[copyFile], :copyFileSync #object[copyFileSync], :createReadStream #object[createReadStream], :createWriteStream #object[createWriteStream], :exists #object[exists], :existsSync #object[existsSync], :fchown #object[fchown], :fchownSync #object[fchownSync], :fchmod #object[fchmod], :fchmodSync #object[fchmodSync], :fdatasync #object[fdatasync], :fdatasyncSync #object[fdatasyncSync], :fstat #object[fstat], :fstatSync #object[fstatSync], :fsync #object[fsync], :fsyncSync #object[fsyncSync], :ftruncate #object[ftruncate], :ftruncateSync #object[ftruncateSync], :futimes #object[futimes], :futimesSync #object[futimesSync], :lchown #object[lchown], :lchownSync #object[lchownSync], :lchmod nil, :lchmodSync nil, :link #object[link], :linkSync #object[linkSync], :lstat #object[lstat], :lstatSync #object[lstatSync], :mkdir #object[mkdir], :mkdirSync #object[mkdirSync], :mkdtemp #object[mkdtemp], :mkdtempSync #object[mkdtempSync], :open #object[open], :openSync #object[openSync], :readdir #object[readdir], :readdirSync #object[readdirSync], :read #object[read], :readSync #object[readSync], :readFile #object[readFile], :readFileSync #object[readFileSync], :readlink #object[readlink], :readlinkSync #object[readlinkSync], :realpath #object[realpath], :realpathSync #object[realpathSync], :rename #object[rename], :renameSync #object[renameSync], :rmdir #object[rmdir], :rmdirSync #object[rmdirSync], :stat #object[stat], :statSync #object[statSync], :symlink #object[symlink], :symlinkSync #object[symlinkSync], :truncate #object[truncate], :truncateSync #object[truncateSync], :unwatchFile #object[unwatchFile], :unlink #object[unlink], :unlinkSync #object[unlinkSync], :utimes #object[utimes], :utimesSync #object[utimesSync], :watch #object[watch], :watchFile #object[watchFile], :writeFile #object[writeFile], :writeFileSync #object[writeFileSync], :write #object[write], :writeSync #object[writeSync], :Dirent #object[Dirent], :Stats #object[Stats], :ReadStream #object[ReadStream], :WriteStream #object[WriteStream], :FileReadStream #object[ReadStream], :FileWriteStream #object[WriteStream], :_toUnixTimestamp #object[toUnixTimestamp], :F_OK 0, :R_OK 4, :W_OK 2, :X_OK 1, :constants #object[Object]}
[1:1]~cljs.user=> (require '["fs" :as fs])
nil
[1:1]~cljs.user=> fs
#js {:appendFile #object[appendFile], :appendFileSync #object[appendFileSync], :access #object[access], :accessSync #object[accessSync], :chown #object[chown], :chownSync #object[chownSync], :chmod #object[chmod], :chmodSync #object[chmodSync], :close #object[close], :closeSync #object[closeSync], :copyFile #object[copyFile], :copyFileSync #object[copyFileSync], :createReadStream #object[createReadStream], :createWriteStream #object[createWriteStream], :exists #object[exists], :existsSync #object[existsSync], :fchown #object[fchown], :fchownSync #object[fchownSync], :fchmod #object[fchmod], :fchmodSync #object[fchmodSync], :fdatasync #object[fdatasync], :fdatasyncSync #object[fdatasyncSync], :fstat #object[fstat], :fstatSync #object[fstatSync], :fsync #object[fsync], :fsyncSync #object[fsyncSync], :ftruncate #object[ftruncate], :ftruncateSync #object[ftruncateSync], :futimes #object[futimes], :futimesSync #object[futimesSync], :lchown #object[lchown], :lchownSync #object[lchownSync], :lchmod nil, :lchmodSync nil, :link #object[link], :linkSync #object[linkSync], :lstat #object[lstat], :lstatSync #object[lstatSync], :mkdir #object[mkdir], :mkdirSync #object[mkdirSync], :mkdtemp #object[mkdtemp], :mkdtempSync #object[mkdtempSync], :open #object[open], :openSync #object[openSync], :readdir #object[readdir], :readdirSync #object[readdirSync], :read #object[read], :readSync #object[readSync], :readFile #object[readFile], :readFileSync #object[readFileSync], :readlink #object[readlink], :readlinkSync #object[readlinkSync], :realpath #object[realpath], :realpathSync #object[realpathSync], :rename #object[rename], :renameSync #object[renameSync], :rmdir #object[rmdir], :rmdirSync #object[rmdirSync], :stat #object[stat], :statSync #object[statSync], :symlink #object[symlink], :symlinkSync #object[symlinkSync], :truncate #object[truncate], :truncateSync #object[truncateSync], :unwatchFile #object[unwatchFile], :unlink #object[unlink], :unlinkSync #object[unlinkSync], :utimes #object[utimes], :utimesSync #object[utimesSync], :watch #object[watch], :watchFile #object[watchFile], :writeFile #object[writeFile], :writeFileSync #object[writeFileSync], :write #object[write], :writeSync #object[writeSync], :Dirent #object[Dirent], :Stats #object[Stats], :ReadStream #object[ReadStream], :WriteStream #object[WriteStream], :FileReadStream #object[ReadStream], :FileWriteStream #object[WriteStream], :_toUnixTimestamp #object[toUnixTimestamp], :F_OK 0, :R_OK 4, :W_OK 2, :X_OK 1, :constants #object[Object]}
[1:1]~cljs.user=>
```

**Comment by chriswatrous** [comment]
It works for me when the repl is started with `npx shadow-cljs node-repl`. I'll try to make a more detailed reproduction

**Comment by chriswatrous** [comment]
**shadow-cljs.edn**
```
{:source-paths ["src/cljs"]
 :builds {:app {:target :node-script
                :output-to "target/main.js"
                :main myapp.core/main
                :devtools {:after-load server.main/reload!}}}}
```

**src/cljs/myapp/core.cljs**
```
(ns myapp.core)

; this works
(println (js/Object.keys (js/require "cluster")))

(defn main []
  ; this works
  (println (js/Object.keys (js/require "cluster"))))

(def js-require js/require)

(comment
  ; eval this after connecting to switch to cljs repl
  (shadow.cljs.devtools.api/nrepl-select :app)

  ; this fails
  (println (js/Object.keys (js/require "cluster")))

  ; this works
  (println (js/Object.keys (js-require "cluster"))))
```

steps
- `npx shadow-cljs watch app`
- in a different terminal window `node target/main.js`
- connect proto-repl
- eval the things in the comment form

**Comment by thheller** [comment]
I don't have proto-repl and can't test that.

I can understand why your "cheat" works since `require` is not a "global" function regularly so you wouldn't find it on `global`. The REPL should however have access to it normally. not sure why it doesn't given that `node-repl` works.

Can you try `(require '["cluster" :as x])` then `x`?

**Comment by chriswatrous** [comment]
If I eval `(require '["cluster" :as x])` in the editor, x is nil.

If I put that outside of the comment block, compile error,

If I put `(:require ["cluster" :as x]))` in the ns form, x is the correct object.

**Comment by thheller** [comment]
Should be fixed in `2.8.41`.
